src/biogeophys/HydrologyNoDrainageMod.F90

```diff
@@ -650,6 +659,29 @@ subroutine HydrologyNoDrainage(bounds, &
 
     end associate
 
+
+  contains
+    
+    ! Subroutines for storing the time derivative of top most soil layer moisture. This is
+    ! used for diagnosing the downwards moisture flux within FAN.
+
+    subroutine store_tsl_moisture(waterstatebulk_inst)
+      type(waterstatebulk_type), intent(inout) :: waterstatebulk_inst
+      associate(h2osoi_tend_tsl => waterstatebulk_inst%h2osoi_tend_tsl_col(bounds%begc:bounds%endc), &
+           h2osoi_liq_tsl => waterstatebulk_inst%h2osoi_liq_col(bounds%begc:bounds%endc,1))
+        h2osoi_tend_tsl = h2osoi_liq_tsl
+      end associate
+    end subroutine store_tsl_moisture
+
+    subroutine eval_tsl_moist_tend(waterstatebulk_inst)
+      type(waterstatebulk_type), intent(inout) :: waterstatebulk_inst
+      associate(h2osoi_tend_tsl => waterstatebulk_inst%h2osoi_tend_tsl_col(bounds%begc:bounds%endc), &
+           h2osoi_liq_tsl => waterstatebulk_inst%h2osoi_liq_col(bounds%begc:bounds%endc,1))
+        h2osoi_tend_tsl = (h2osoi_liq_tsl - h2osoi_tend_tsl) / dtime
```

Comment:
This should also be done in a filter loop.

Reply:
I changed these to filter loops over filter_hydrologyc, I guess that's OK?

Reply:
Which filter to use depends on how this variable is used elsewhere in the code, which I haven't checked. If the hydrologyc filter agrees with the filter used in other places where this variable is used, then it's the right filter to use.

Reply:
FAN actually uses the soil column filter, but I was thinking more about consistency with other hydrology-related variables.

Reply:
Okay, thanks. Since the hydrology filter contains everything in the soil filter and just a bit more, either one seems fine to me.